### PR TITLE
fix: improper shuffling of records with same position in aln subcommand

### LINKED ISSUE #76: Subsampling an alignment file results in "user-provided comparison function does not correctly implement a total order" error
When trying to run:

```bash
export RUST_BACKTRACE=full
rasusa aln --coverage 30 HG002.bam
```

I get the following error:

```
[2025-01-04T00:49:23Z INFO ] Subsampling alignment file: "HG002.bam"
[2025-01-04T00:49:23Z INFO ] Using seed: 16213766130575630274
[2025-01-04T00:49:23Z INFO ] Writing subsampled alignment to stdout
[2025-01-04T00:49:23Z INFO ] Subsampling chromosome: chr1
thread 'main' panicked at core/src/slice/sort/shared/smallsort.rs:865:5:
user-provided comparison function does not correctly implement a total order
stack backtrace:
   0:     0x5555558e086a - <unknown>
   1:     0x5555559035c3 - <unknown>
   2:     0x5555558ddf93 - <unknown>
   3:     0x5555558e06b2 - <unknown>
   4:     0x5555558e165c - <unknown>
   5:     0x5555558e14a2 - <unknown>
   6:     0x5555558e1c37 - <unknown>
   7:     0x5555558e1a96 - <unknown>
   8:     0x5555558e0d49 - <unknown>
   9:     0x5555558e175c - <unknown>
  10:     0x5555555b5780 - <unknown>
  11:     0x5555555b5bab - <unknown>
  12:     0x5555555eb64e - <unknown>
  13:     0x5555555d4a55 - <unknown>
  14:     0x5555555d4fa8 - <unknown>
  15:     0x5555555d4fa8 - <unknown>
  16:     0x5555555d4fa8 - <unknown>
  17:     0x5555555d4fa8 - <unknown>
  18:     0x5555555d4fa8 - <unknown>
  19:     0x5555555d4fa8 - <unknown>
  20:     0x5555555d4fa8 - <unknown>
  21:     0x5555555f515f - <unknown>
  22:     0x5555555ee134 - <unknown>
  23:     0x5555555ca320 - <unknown>
  24:     0x5555555ef0d6 - <unknown>
  25:     0x5555555e7963 - <unknown>
  26:     0x5555555ec72d - <unknown>
  27:     0x5555558d814e - <unknown>
  28:     0x5555555ef2e5 - <unknown>
  29:     0x2aaaab61b555 - __libc_start_main
  30:     0x5555555b5ee0 - <unknown>
  31:                0x0 - <unknown>
```

I am using version `rasusa-aln 2.1.0` 

Any idea what might be causing this error?

### PR DESCRIPTION
**Pull request by mbhall88**
The random_sort function was only introducing randomness at the comparison level rather than properly shuffling groups of records with the same position. This meant that records with identical positions would maintain their original relative order instead of being randomly distributed within their group.

Changes:
- Replace random_sort/random_compare with shuffle_records_by_position
- Group records by position before shuffling within each group
- Add comprehensive tests including edge cases and regression tests
- Fix clippy format string warnings in alignment.rs, fastx.rs, and subsampler.rs
- Add demonstration test that reproduces the original bug

The new implementation ensures that records with the same genomic position are properly randomized while maintaining correct positional ordering between different positions, fixing the reproducible bias in subsampling results.

Fixes #76

### CODE REVIEW

File: src/alignment.rs [hunk 1]
```diff
@@ -1,5 +1,5 @@
 use std::borrow::Cow;
-use std::cmp::{Ordering, Reverse};
+use std::cmp::Reverse;
```
[the remark this answers is not in the document]
Comment (mbhall88) [reply src/alignment.rs:2]: Well given the fact that all the CI checks pass, this doesn't cause a compilation error
